source/extensions/matching/input_matchers/cel_matcher/matcher.cc

```diff
@@ -0,0 +1,50 @@
+#include "source/extensions/matching/input_matchers/cel_matcher/matcher.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Matching {
+namespace InputMatchers {
+namespace CelMatcher {
+
+CelInputMatcher::CelInputMatcher(const CelExpression& input_expr) {
+  expr_builder_ = Extensions::Filters::Common::Expr::createBuilder(nullptr);
+  switch (input_expr.expr_specifier_case()) {
+  case CelExpression::ExprSpecifierCase::kParsedExpr:
+    compiled_expr_ =
+        Filters::Common::Expr::createExpression(*expr_builder_, input_expr.parsed_expr().expr());
+    return;
+  case CelExpression::ExprSpecifierCase::kCheckedExpr:
+    compiled_expr_ =
+        Filters::Common::Expr::createExpression(*expr_builder_, input_expr.checked_expr().expr());
```

Comment:
Compiled expression reference data from the underlying protobuf - the protobuf must be stored along with the compiled expression.

Reply:
hmm, I did see the comment in CEL lib that expr must outlive the resulting CelExpression before. But i was thinking it should cause issue(like if builder not outlive CelExpression) if i don't do that. And it runs fine so far.

I agree we are better off to store it but I am curious where does expression reference the data?